testctrl/proto/test_sessions.proto

```diff
@@ -0,0 +1,165 @@
+// Copyright 2020 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package proto;
+
+option go_package = "github.com/grpc/grpc/testctrl/genproto/testctrl/svc";
+
+import "google/protobuf/duration.proto";
+import "google/protobuf/timestamp.proto";
+import "google/longrunning/operations.proto";
+import "src/proto/grpc/testing/control.proto";
+
+service TestSessions {
+  // Starts a set of test scenarios.
+  //
+  // This is a long-running operation that is managed by the Operations service
+  // on the same server.  The long-running operation will assign a unique
+  // identifier to the session.
+  //
+  // The unique identifier can be used to poll for the session's status and
+  // results.  Streaming is not supported at present.
+  //
+  // It can also be used to cancel while in progress, but cancellations operate
+  // like killing a running process. If they occur while results are being
+  // reported, the persistent storage may receive some but not all of them.
+  //
+  // For the specification of a google.longrunning.Operation message, see the
+  // Long-running AIP at https://aip.dev/151.
+  rpc StartTestSession(StartTestSessionRequest) returns (google.longrunning.Operation) {
+    option (google.longrunning.operation_info) = {
+      response_type: "TestSessionResults"
+      metadata_type: "TestSessionMetadata"
+    };
+  }
+}
+
+message Container {
+  enum Kind {
+    // No kind was specified, so the service will attempt to infer one.
+    //
+    // This inference works as follows:
+    //   1. If this field belongs to the driver_container of the
+    //      StartTestSessionRequest, this kind is inferred to be a driver.
+    //   2. Otherwise, this field is inferred to be a client worker.
+    UNSPECIFIED = 0;
+
+    // Orchestrates the test amongst workers and reports results.
+    DRIVER = 1;
+
+    // Processes incoming requests from client containers.
+    SERVER = 2;
+
+    // Sends outgoing requests to a server container.
+    CLIENT = 3;
+  }
+
+  // The name of the image as it appears in the registry.
+  //
+  // When using GCR, this will likely be a fully-qualified URL. For example,
+  // the name of the Java worker may be "gcr.io/grpc-testing/java_qps_worker".
+  string name = 1;  // [(google.api.field_behavior) = REQUIRED];
```

Comment:
At launch of the prototype, images will be restricted to the `gcr.io/grpc-testing/*` to ensure random images and non-trusted registries are not used. Any image outside this scope will result in a bad request.